src/modules/navigator/GeofenceBreachAvoidance/geofence_breach_avoidance.cpp

```diff
@@ -182,7 +197,16 @@ GeofenceBreachAvoidance::generateLoiterPointForMultirotor(geofence_violation_typ
 
 	} else if (violation_type.flags.dist_to_home_exceeded) {
 
-		return waypointFromHomeToTestPointAtDist(math::max(_max_hor_dist_home - _min_hor_dist_to_fence_mc, 0.0f));
+		// This is where the Horizontal Hold position is computed.
+		// Originally, the argument was set to the Geofence radius (_max_hor_dist_home) - a minimum distance that the drone could begin braking from based on acceleration/jerk limit parameters (_min_hor_dist_to_fence_mc).
+		// The intention was to ensure HOLD was just inside the Geofence.
+
+		// Instead, we want to trigger a breach at Geofence but set Hold to the braking distance which is outside the Geofence to preserve the controlled deceleration and prevent overshoot corrections.
+		// Hence, the argument instead uses the Geofence Radius + braking distance from that radius.
+
+		// The braking distance was arbitrarily divided by 2 as it was still overkill.
+		// (was overshooting by ~30m on a 30m geofence, i.e holding 60m from home)
+		return waypointFromHomeToTestPointAtDist((computeBrakingDistanceMultirotor() / 2) + _max_hor_dist_home);
```

Comment:
As discussed on the phone - a couple of suggestions here.  Worth looking at both I think.

**Suggestion 1 - improve "Predictor"**
Have a look at the behaviour of "Predictor" (lines 245 & 260 of geofence_breach_avoidance.cpp).  This uses various PX4 params to do what looks like a sensible stopping distance estimate.  My suspicion is the params it is using ("param_mpc_acc_hor", "param_mpc_jerk_auto", "param_mpc_acc_down_max" etc.) may be set too conservatively, but do check where they are used and if they affect any other flight behaviour before altering.  

If not used elsewhere in our normal flight control, then it might just be as simple as modifying those params to fix the overly conservative stopping distance estimate - rather than having to fudge it with a divide by 2.

**Suggestion 2 - Change the loiter point to avoid the generateLoiterPointForMultirotor method entirely**
The generateLoiterPointForMultirotor is mostly about trying to stop you on the geofence boundary, which is no longer your aim now that you are triggering on the boundary.
Hence you could simplify the loiter point definition to just be the predicted braking distance ahead of the drone, with something like this on Line 831 of navigator_main.cpp (do check I've got the right members - e.g. "loiter_altitude_amsl = fence_violation_test_point.alt;" may be wrong):
```
if (_vstatus.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
	// the computation of the braking distance does not match the actual braking distance. Until we have a better model
	// we set the loiter point to the current position, that will make sure that the vehicle will loiter inside the fence
	if (sees_geofence_behaviour) {
	// Sees Behaviour is to trigger on boundary - so we just try to stop at the test point (which is the predicted braking distance ahead of the multirotor current location).
		lointer_center_lat_lon = matrix::Vector2<double>(fence_violation_test_point.lat, fence_violation_test_point.long);
		loiter_altitude_amsl = fence_violation_test_point.alt;

	} else {
	// PX4 default behaviour is to try and stop on the boundary using the generateLoiterPointForMultirotor method
		lointer_center_lat_lon =  _gf_breach_avoidance.generateLoiterPointForMultirotor(gf_violation_type,
					&_geofence);

		loiter_altitude_amsl = _gf_breach_avoidance.generateLoiterAltitudeForMulticopter(gf_violation_type);
	}

} 
```